programs/loader-v4/src/lib.rs

```diff
@@ -575,26 +585,18 @@ pub fn process_instruction_inner(
         .map_err(|err| Box::new(err) as Box<dyn std::error::Error>)
     } else {
         let program = instruction_context.try_borrow_last_program_account(transaction_context)?;
-        if !loader_v4::check_id(program.get_owner()) {
```

Comment:
Could you explain why this check is not needed?

Reply:
The program runtime invokes programs by their owner, so this is implicitly always true here.